frontend/src/metabase/admin/datamodel/metadata/components/MetadataHeader/MetadataHeader.tsx

```diff
@@ -53,60 +53,67 @@ const MetadataHeader = ({
   selectedTableId,
   onSelectDatabase,
 }: MetadataHeaderProps) => {
+  const [isFieldOrderSidesheetOpen, setIsFieldOrderSidesheetOpen] =
+    useState(false);
+
   useLayoutEffect(() => {
     if (databases.length > 0 && selectedDatabaseId == null) {
       onSelectDatabase(databases[0].id, { useReplace: true });
     }
   }, [databases, selectedDatabaseId, onSelectDatabase]);
 
   return (
-    <div
-      data-testid="admin-metadata-header"
-      className={cx(
-        "MetadataEditor-header",
-        CS.flex,
-        CS.alignCenter,
-        CS.flexNoShrink,
-        CS.py4,
-      )}
-    >
-      <Icon
-        className={cx(CS.flex, CS.alignCenter, CS.flexNoShrink, CS.textMedium)}
-        name="database"
-      />
-      <div className={cx("MetadataEditor-headerSection", CS.h2)}>
-        <DatabaseDataSelector
-          databases={databases}
-          selectedDatabaseId={selectedDatabaseId}
-          setDatabaseFn={onSelectDatabase}
-          style={{ padding: 0, paddingLeft: 8 }}
-        />
-      </div>
+    <Flex align="center" data-testid="admin-metadata-header" flex="1" py="xl">
+      <Flex align="center" gap="sm">
+        <Text c="text-medium" display="flex" flex="0 0 auto">
+          <Icon name="database" />
+        </Text>
+
+        <Text fw="bold" size="xl">
+          <DatabaseDataSelector
+            className={S.databaseDataSelectors}
+            databases={databases}
+            selectedDatabaseId={selectedDatabaseId}
+            setDatabaseFn={onSelectDatabase}
+          />
+        </Text>
+      </Flex>
+
       {selectedDatabaseId && selectedSchemaId && selectedTableId && (
-        <div
-          className={cx(
-            "MetadataEditor-headerSection",
-            CS.flex,
-            CS.flexAlignRight,
-            CS.alignCenter,
-            CS.flexNoShrink,
-          )}
+        <Flex
+          align="center"
+          flex="1 0 auto"
+          gap="md"
+          justify="flex-end"
+          mx="xl"
         >
-          <span className={cx(CS.ml4, CS.mr3)}>
-            <TableSettingsLink
-              aria-label={t`Settings`}
-              to={Urls.dataModelTableSettings(
-                selectedDatabaseId,
-                selectedSchemaId,
-                selectedTableId,
-              )}
-            >
-              <Icon name="gear" />
-            </TableSettingsLink>
-          </span>
-        </div>
+          <Button
+            aria-label={t`Edit column order`}
+            leftSection={<Icon name="sort_arrows" />}
+            onClick={() => setIsFieldOrderSidesheetOpen((isOpen) => !isOpen)}
```

Comment:
Maybe using [`useDisclosure`](https://mantine.dev/hooks/use-disclosure/)'s `toggle` would be easier here.

Reply:
Good idea! 2b1663fb9dcad65df8abb06f46bc7c3c81b5fe26